common/djangoapps/student/management/commands/bulk_change_enrollment_csv.py

```diff
@@ -0,0 +1,111 @@
+                except:
+                    raise CommandError('Invalid or non-existant user {}'.format(row['user']))
+
+                try:
+                    course_enrollment = CourseEnrollment.get_enrollment(user, course_key)
+                    #  If student is not enrolled in course enroll the student in free mode
+                    if not course_enrollment:
+                        # try to create a enroll user in default course enrollment mode in case of
+                        # professional it will break because of no default course mode.
+                        try:
+                            course_enrollment = CourseEnrollment.get_or_create_enrollment(user=user,
+                                                                                          course_key=course_key)
+                        except Exception:  # pylint: disable=broad-except
+                            # In case if no free mode is available.
+                            course_enrollment = None
+                    # if student already had a enrollment and its mode is same as the provided one
+                    if course_enrollment and course_enrollment.mode == row['mode']:
+                        logger.info("Student [%s] is already enrolled in Course [%s] in mode [%s].", user.username,
+                                    course_key, course_enrollment.mode)
+                        # set the enrollment to active if its not already active.
+                        if not course_enrollment.is_active:
+                            course_enrollment.is_active = True
+                        course_enrollment.save()
+                    elif course_enrollment:
+                        # if student enrollment exists update it to new mode.
+                        with transaction.atomic():
+                            course_enrollment.update_enrollment(
+                                mode=row['mode'],
+                                is_active=True,
+                                skip_refund=True
+                            )
+                            course_enrollment.save()
+                            enrollment_attrs = []
+                            if row['mode'] == 'credit':
+                                enrollment_attrs.append({
+                                    'namespace': 'credit',
+                                    'name': 'provider_id',
+                                    'value': course_key.org,
+                                })
+                                CourseEnrollmentAttribute.add_enrollment_attr(enrollment=course_enrollment,
+                                                                              data_list=enrollment_attrs)
+                    else:
+                        # if student enrollment do not exists directly enroll in new mode.
+                        CourseEnrollment.enroll(user=user, course_key=course_key, mode=row['mode'])
```

Comment:
is not this same as what we are doing at line 74 ?

Reply:
explained in person